src/main/java/com/facebook/openwifirrm/Launcher.java

```diff
@@ -82,21 +82,24 @@ private RRMConfig readRRMConfig(File configFile) throws IOException {
 			// Read file
 			logger.info("Reading config file '{}'", configFile.getPath());
 			String contents = Utils.readFile(configFile);
-			JSONObject userConfig = new JSONObject(contents);
-
-			// In case of any added/missing values, we want to build off the
-			// defaults in RRMConfig, so this code gets more complex...
-			JSONObject fullConfig =
-				new JSONObject(gson.toJson(new RRMConfig()));
-			Utils.jsonMerge(fullConfig, userConfig);
-			config = gson.fromJson(fullConfig.toString(), RRMConfig.class);
-
-			// Compare merged config with contents as read from disk
-			// If any differences (ex. added fields), overwrite config file
-			if (!fullConfig.toString().equals(userConfig.toString())) {
-				logger.info("Rewriting config file with new changes...");
-				try (Writer writer = new FileWriter(configFile)) {
-					gson.toJson(config, writer);
+			if (contents == null || contents.isBlank()) {
```

Comment:
same here, flip the outer conditions but you'll need to do the `contents.isEmpty()` check yourself (`new JSONObject` doesn't handle it)